### PR TITLE
docs(*): add pull request review guide

### LINKED ISSUE #1434: document how to review a PR

### PR DESCRIPTION
**Pull request by michelleN**
+ add README in docs dir
+ remove redundant demo
+ rename dev_guide dir to development_guide

resolves #1434

Signed-off-by: Michelle Noorali <minooral@microsoft.com>

Please mark with X for applicable areas.

-->
**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ X] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [ ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? No
    -   Did you notify the maintainers and provide attribution? NA

2. Is this a breaking change? No

### CODE REVIEW

File: docs/pull_request_review_guide.md
Comment (shashankram) [inline docs/pull_request_review_guide.md:16]: Squashing is not required anymore. PR authors can request maintainers to squash their commits before merging. We established that this is standard practice across many public projects, including Envoy, Contour etc.
https://github.com/openservicemesh/osm/blob/main/CONTRIBUTING.md#merging-pull-requests

Comment (michelleN) [reply docs/pull_request_review_guide.md]: thanks. I updated the text to reflect this. Please let me know if you have a different suggestion.

**Review by shashankram — APPROVED** [review]

**Review by shalier — APPROVED** [review]

**Pull request merged**
